### ISSUE TITLE
Freezing the Discriminator 

### ISSUE THREAD

**Issue by mohjaba** [roatienza/Deep-Learning-Experiments#9]
Hi, thank you very much for this wonderful example for implementing GAN using Keras.
I have one question, based on what I understood from the theory that there are three stages GAN should be train with:
1. training D for ground truth data as input with label 1 as output
2. training D using G output as an input and label  0 as output
3. same as (2) but with freezing D weights and label 1 as D output
my question is: I didn't see any freezing step in your code. So, is that deferent way of implementation? or I miss something?
Thanks again

**Comment by roatienza** [comment]
In some GAN training, they freeze the weights of the discriminator. Here, the discrimimator and generator are trained simultaneously.
